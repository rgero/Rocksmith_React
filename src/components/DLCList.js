import React from 'react';
import { connect } from 'react-redux';

import _ from 'lodash';

// Import React Table
import ReactTable from 'react-table';
import 'react-table/react-table.css';

export const DLCList = (props) => {
    var data = props.dlc;
    return (
        <div className='content-container'>
            <ReactTable
                data={data}
                columns={[
                    {
                    Header: 'DLC',
                    columns: [
                        {
                        Header: 'Artist Name',
                        accessor: 'artist'
                        },
                        {
                            Header: 'Song Name',
                            accessor: 'name'
                        },
                        {
                            Header: 'Lead',
                            accessor: 'leadTuning'
                        },
                        {
                            Header: 'Rhythm',
                            accessor: 'rhythmTuning'
                        },
                        {
                            Header: 'Bass',
                            accessor: 'bassTuning'
                        }
                    ]
                    }
                ]}
                defaultPageSize={10}
                className='-striped -highlight'
                />
        </div>
    )
};

const mapStateToProps = (state, props)=>{
    return {
        dlc: state
    };
}

export default connect(mapStateToProps)(DLCList);