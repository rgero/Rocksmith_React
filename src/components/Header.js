import React from 'react';
import { Link } from 'react-router-dom';


export const Header = () => (
    <header className='header'>
        <div className='header__content'>
            <Link className='header__title' to='/' exact='true'><h1>Rocksmith DLC Searcher</h1></Link>
        </div>
    </header>
);

export default Header;