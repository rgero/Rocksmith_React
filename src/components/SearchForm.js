import React from 'react';

export class SearchForm extends React.Component {

    constructor(props){
        super(props);

        //Functions
        this.onSubmit = this.onSubmit.bind(this);
        this.onTextChange = this.onTextChange.bind(this);
        this.resetState = this.resetState.bind(this);

        this.state = {
            artistNames: '',
            songNames: '',
            leadTunings: '',
            rhythmTunings: '',
            bassTunings: ''
        };
    };
    
    onSubmit(e){
        e.preventDefault();
        this.props.onSubmit({
            artistNames: this.state.artistNames,
            songNames: this.state.songNames,
            leadTunings: this.state.leadTunings,
            rhythmTunings: this.state.rhythmTunings,
            bassTunings: this.state.bassTunings
        })
    }

    resetState(){
        this.setState({
            artistNames: '',
            songNames: '',
            leadTunings: '',
            rhythmTunings: '',
            bassTunings: '',
        })
    }

    onTextChange = (key) => (e) => {
        var newValue = e.target.value;
        switch(key){
            case 'artistNames':
                this.setState({
                    artistNames: newValue
                });
                break;
            case 'songNames':
                this.setState({
                    songNames: newValue
                });
                break;
            case 'leadTunings':
                this.setState({
                    leadTunings: newValue
                });
                break;
            case 'rhythmTunings':
                this.setState({
                    rhythmTunings: newValue
                });
                break;
            case 'bassTunings':
                this.setState({
                    bassTunings: newValue
                });
                break;
            default:
                break;
        }
    }

    render() {
        return (
            <div className='content-container'>
                <form className='form' onSubmit={this.onSubmit}>
                    <div className='form__input'>
                        <label>Artist Names</label>
                        <input
                            type='text'
                            className='text-input'
                            placeholder='Names'
                            autoFocus
                            value={this.state.artistNames}
                            onChange={this.onTextChange('artistNames')}
                        />
                    </div>
                    <div className='form__input'>
                        <label>Song Names</label>
                        <input
                            type='text'
                            className='text-input'
                            placeholder='Names'
                            autoFocus
                            value={this.state.songNames}
                            onChange={this.onTextChange('songNames')}
                        />
                    </div>
                    <div className='form__input'>
                        <label>Lead Tuning</label>
                        <input
                            type='text'
                            className='text-input'
                            placeholder='Lead Tuning'
                            autoFocus
                            value={this.state.leadTunings}
                            onChange={this.onTextChange('leadTunings')}
                        />
                    </div>
                    <div className='form__input'>
                        <label>Rhythm Tunings</label>
                        <input
                            type='text'
                            className='text-input'
                            placeholder='Rhythm Tunings'
                            autoFocus
                            value={this.state.rhythmTunings}
                            onChange={this.onTextChange('rhythmTunings')}
                        />
                    </div>
                    <div className='form__input'>
                        <label>Bass Tuning</label>
                        <input
                            type='text'
                            className='text-input'
                            placeholder='Bass Tuning'
                            autoFocus
                            value={this.state.bassTunings}
                            onChange={this.onTextChange('bassTunings')}
                        />
                    </div>
                    <div className='buttons'>
                        <button className='button' onClick={this.resetState}>Reset</button>
                        <button className='button'>Submit</button>
                    </div>
                </form>
            </div>
        )
    }

}


export default SearchForm;