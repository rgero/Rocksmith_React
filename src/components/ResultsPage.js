import React, { Component } from 'react';

//Components
import Header from './Header';
import DLCList from './DLCList';

export class ResultsPage extends Component {
  render() {
    return (
      <div>
        <Header/>
        <DLCList/>
      </div>
    );
  }
}

export default ResultsPage;
