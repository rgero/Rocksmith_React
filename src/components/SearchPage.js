import React from 'react';
import {connect} from 'react-redux';
//Components
import Header from './Header';
import SearchHeader from './SearchHeader';
import { SearchForm } from './SearchForm';

class SearchPage extends React.Component {

  constructor(props){
    super(props);
    this.onSubmit = this.onSubmit.bind(this);

  }

  onSubmit(query){
    this.props.onSubmit(query);
    this.props.history.push('/results')
  }

  render() {
    return (
      <div>
        <Header/>
        <SearchHeader/>
        <SearchForm onSubmit={this.onSubmit}/>
      </div>
    );
  }
}

export default connect(undefined, undefined)(SearchPage);
