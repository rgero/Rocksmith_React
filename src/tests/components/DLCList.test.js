// react-test-renderer
import React from 'react'
import {shallow} from 'enzyme'
import {DLCList} from '../../components/DLCList';

import dlcData from '../fixtures/dlc';

test('DLCList Rendering Test - No Data Rendering', ()=>{
    const wrapper = shallow(<DLCList/>);
    expect(wrapper).toMatchSnapshot();
})

test('DLCList Rendering Test - Data Rendering', ()=>{
    const wrapper = shallow(<DLCList dlc={dlcData}/>);
    expect(wrapper).toMatchSnapshot();
})
