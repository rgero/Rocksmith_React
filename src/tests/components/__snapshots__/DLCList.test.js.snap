// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DLCList Rendering Test - Data Rendering 1`] = `
<div
  className="content-container"
>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highlight"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "Placeholder": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minResizeWidth": 11,
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Header": "DLC",
          "columns": Array [
            Object {
              "Header": "Artist Name",
              "accessor": "artist",
            },
            Object {
              "Header": "Song Name",
              "accessor": "name",
            },
            Object {
              "Header": "Lead",
              "accessor": "leadTuning",
            },
            Object {
              "Header": "Rhythm",
              "accessor": "rhythmTuning",
            },
            Object {
              "Header": "Bass",
              "accessor": "bassTuning",
            },
          ],
        },
      ]
    }
    data={
      Array [
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (AEADF#B)",
          "name": "Guardians of Asgaard",
          "rhythmTuning": "Custom Tuning (AEADF#B)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "The Pursuit of VIkings",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "War Of The Gods",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "Twilight Of The Thunder God",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Amon Amarth",
          "bassTuning": "E Standard",
          "leadTuning": "Custom Tuning (B Standard)",
          "name": "Death In Fire",
          "rhythmTuning": "Custom Tuning (B Standard)",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Bloodlines",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Murmaider",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Black Fire Upon Us",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Go Into The Water",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Awaken",
          "rhythmTuning": "C Standard",
        },
        Object {
          "artist": "Dethklok",
          "bassTuning": "C Standard",
          "leadTuning": "C Standard",
          "name": "Thunderhorse",
          "rhythmTuning": "C Standard",
        },
      ]
    }
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPage={0}
    defaultPageSize={10}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    multiSort={true}
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onFetchData={[Function]}
    originalKey="_original"
    pageJumpText="jump to page"
    pageSizeOptions={
      Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ]
    }
    pageText="Page"
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    resolveData={[Function]}
    rowsSelectorText="rows per page"
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={true}
    style={Object {}}
    subRowsKey="_subRows"
  />
</div>
`;

exports[`DLCList Rendering Test - No Data Rendering 1`] = `
<div
  className="content-container"
>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highlight"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "Placeholder": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minResizeWidth": 11,
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Header": "DLC",
          "columns": Array [
            Object {
              "Header": "Artist Name",
              "accessor": "artist",
            },
            Object {
              "Header": "Song Name",
              "accessor": "name",
            },
            Object {
              "Header": "Lead",
              "accessor": "leadTuning",
            },
            Object {
              "Header": "Rhythm",
              "accessor": "rhythmTuning",
            },
            Object {
              "Header": "Bass",
              "accessor": "bassTuning",
            },
          ],
        },
      ]
    }
    data={Array []}
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPage={0}
    defaultPageSize={10}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    multiSort={true}
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onFetchData={[Function]}
    originalKey="_original"
    pageJumpText="jump to page"
    pageSizeOptions={
      Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ]
    }
    pageText="Page"
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    resolveData={[Function]}
    rowsSelectorText="rows per page"
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={true}
    style={Object {}}
    subRowsKey="_subRows"
  />
</div>
`;
